### DISCUSSION TITLE
Can and should Fin<T> behave like Validation<Error, T> instead of Either<Error, T>?

### DISCUSSION THREAD

**Discussion by Demmon1** [louthy/language-ext#1364] [Q&A] [answered]
The README for `Fin<T>` states:
> It is functionally exactly the same as Either<Error, A>, it is a convenience type to avoid the generics pain of Either.

However, as far as I can tell, `Validation<E, T>` is functionally the same as `Either<E, T>` except that it collects errors when used as an applicative functor. Since `Error` is monoidal, couldn't `Fin<T>` also collect errors in the same way?

Naively, I would like to choose whether `Fin<T>` collects errors or short-circuits by choosing to use it's applicative or monadic behavior, respectively. I believe this is how `Validation<E, T>` works, but perhaps I'm missing something here. Since the `K<Fin, Func<A, B>>` and `K<F, A>` instances passed to `Applicative<Fin>.Apply` may both contain `Error` instances, it doesn't make sense to me that the `K<F, A>` error would be ignored and thrown out. If I have two `Error` instances, I would like to see both in my result.

Is it indeed possible for `Fin<T>` to collect errors when calling `Applicative<Fin>.Apply`?
If so, should `Fin<T>` be changed to act more like `Validation<Error, T>` or why not?

(If nothing else, I would like to not have to choose between `Fin<T>` and `Validation<Error, T>` in function signatures!)

**Comment by louthy** [comment, marked as answer]
Agreed, it's released in `v5.0.0-beta-15`.

Includes: `Fin<A>`, `FinT<M, A>`, `Try<A>`, and `TryT<M, A>`.
